src/server/modules/dao/ModDAO.ts

```diff
@@ -26,12 +33,29 @@ export default class ModDAO extends BaseDAO<IDbMod> implements IDbModDAO {
                 .split("@")[0]
                 .trim()
         }));
-        const foundDependencies = await await this.collection.find({ $or: _dependencies }).toArray();
-        if (foundDependencies.length !== _dependencies.length) {
+        const foundDependencies = await this.collection.find({ $or: _dependencies }).toArray();
+        const deDupedFound = await this.dedupeDependencies(foundDependencies);
+        if (deDupedFound.length !== _dependencies.length) {
             throw new ServerError("server.invalid_dependencies", [], 400);
         }
+
         return foundDependencies;
     }
+    public async dedupeDependencies(dependencies: Dependencies) {
+        const clone: Dependencies = JSON.parse(JSON.stringify(dependencies));
+
+        const map = new Map<string, string>();
+        for (const dep of clone) {
+            const version = map.get(dep.name);
+            if (version === undefined) {
+                map.set(dep.name, dep.version);
+            } else if (semver.gt(dep.version, version)) {
+                map.set(dep.name, dep.version);
```

Comment:
Since the two branches do the same thing, you could just combine the conditions with an `||`.

In fact, are there ever going to be different versions? The MongoDB query should only return one version, unless 2 requirements were placed on the same mod or something weird. I'd just get rid of the if statement and always do `map.set`.

Reply:
It returns all versions prior to the one requested, that is the basis of the entire bug

Reply:
I think it returns the same version for different gameVersions, which are duplicates of the mod in the database.

Reply:
for when there's a new gameVersion that doesn't break mods I mean

Reply:
Regardless, this de-duper can be applied elsewhere so keeping the logic as-is seems fine to me.

Reply:
sure